source/cgal/intersections.cc

```diff
@@ -685,69 +605,82 @@ namespace CGALWrappers
       }
 
     return vertices;
-
 #  else
-
     Assert(
       false,
       ExcMessage(
         "This function requires a version of CGAL greater or equal than 5.1.5."));
-    (void)cell0;
-    (void)cell1;
-    (void)mapping0;
-    (void)mapping1;
+    (void)vertices0;
+    (void)vertices1;
     (void)tol;
     return {};
 #  endif
   }
 
-
-
   template <>
   std::vector<std::array<Point<3>, 4>>
   compute_intersection_of_cells<3, 3, 3>(
-    const typename Triangulation<3, 3>::cell_iterator &cell0,
-    const typename Triangulation<3, 3>::cell_iterator &cell1,
-    const Mapping<3, 3> &                              mapping0,
-    const Mapping<3, 3> &                              mapping1,
-    const double                                       tol)
+    const std::array<Point<3>, 8> &vertices0,
+    const std::array<Point<3>, 8> &vertices1,
+    const double                   tol)
+  {
+    // TODO:????????!!!
+
+    // Surface_mesh surf0, surf1, sm;
+    // CGALWrappers::dealii_cell_to_cgal_surface_mesh(cell0, mapping0, surf0);
+    // CGALWrappers::dealii_cell_to_cgal_surface_mesh(cell1, mapping1, surf1);
```

Comment:
I can take care of that if you want!

Reply:
That would be great, I couldn't figure out how to do it with just vertices similar to the iterator version because also information on cell orientation is required.

Reply:
Indeed, we need the orientation of the cell to build it. In order to have a version without the iterator, the best option would be to do the same we did for the other intersections.

Reply:
Would you keep the cell iterator version additionally, or scrub it in favor of the version that only needs the vertices?

Reply:
If the version with vertices only works fine, I'd keep only that one. You plan to use only the version with vertices in your MPI implementation, right?

Reply:
Yes, I want to send as little information as possible and don't want to reconstruct cells from vertices after sending. I will share the implementation with you once I have it working. Probably, it would be best to fuse the implementations on collecting quadrature rules before merging to deal.ii :)

Reply:
Ok, I have a working version with vertices only :) I can push it here after your ping.

> it would be best to fuse the implementations on collecting quadrature rules before merging to deal.ii

I agree! So far I have a serial version that collects quadratures on intersections of meshes in one of my branches.

Reply:
Perfect, thank you. It shouldn't take me long to finish up. I just have to wait until the build of deal ii is ready on my machine.

I suggest that you rebase and update the branch after you merged the 3d/3d cuts here and let me know how the branch is called, then I can directly add MPI functionality there :)